### PR TITLE
Feature/filterobs

### PR DESCRIPTION
**Pull request by emanueleemili**
Background filter introduced

### CODE REVIEW

**Comment by andreapiacentini** [comment]
Before issuing a PR toward develop, develop has to be merged into the source branch, conflict solved and test passed again

**Comment by emanueleemili** [comment]
Broken after the merge: results differ significantly on the real assimilation case with 4densvar as well. Something went wrong between the filters and the merge.

File: src/aq/aq_obsvec_mod.F90 [hunk 1]
```diff
@@ -210,6 +211,28 @@ subroutine aq_obsvec_mask_with_missing(self,mask)
...
+type(aq_obsvec),intent(inout) :: self          !< Simulated obs vector
+type(aq_obsvec),intent(inout) :: other         !< Obs vector
+type(aq_obsvec),intent(inout) :: qcflag        !< Mask
+type(fckit_Configuration),intent(in) :: config !< Filter config
+
+real(aq_real) :: threshold
+
+call config%get_or_die("threshold",threshold)
+if ((self%nobs/=other%nobs).or.(self%nlev/=other%nlev)) call abor1_ftn('aq_obsvec_mask: inconsistent sizes')
+write(*,*) 'threshold', threshold
```
Comment (andreapiacentini) [inline src/aq/aq_obsvec_mod.F90:228]: Was it a working print or should it be kept and routed to %info() ?

File: test/testinput/3densvar.yaml [hunk 2]
```diff
@@ -79,6 +79,9 @@ cost function:
       obserrors:
         type: percent
         value: 30
+    obs filters:
```
Comment (andreapiacentini) [inline test/testinput/3densvar.yaml:82]: should we keep a test with filters and one not or the 4densvar is acting as a non filtered test ?

**Review by andreapiacentini — CHANGES_REQUESTED** [review]

File: src/aq/BackgroundCheck.cc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    oops::Log::info() << "BackgroundCheck::postFilter operations " << std::endl;
+    eckit::LocalConfiguration filterConfig(options_.toConfiguration());
+    // Qui devo avere una cosa del genere
+    // obsdb_.getdb("ObsValue", yobs);
+    oops::Log::info() << filterConfig << std::endl;
+    aq_obsvec_threshold_check_f90(hofx.toFortran(), yobs.toFortran(),
+      (*qcflags_).toFortran(), filterConfig);
+    size_t inflate = 0;
+    size_t ireject = 0;
+    // for (size_t jj = 0; jj < yobs.size(); ++jj) {
```
[the remark this answers is not in the document]
Comment (emanueleemili) [reply src/aq/BackgroundCheck.cc]: Removed

File: src/aq/BackgroundCheck.cc [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    //       (*obserr_)[jj] *= *options_.inflation.value();
+    //       ++inflate;
+    //     // or reject observation
+    //     } else {
+    //       (*qcflags_)[jj] = 1;
+    //       ++ireject;
+    //     }
+    //   }
+    // }
+    oops::Log::info() << "BackgroundCheck::postFilter rejected = " << ireject
```
[the remark this answers is not in the document]
Comment (emanueleemili) [reply src/aq/BackgroundCheck.cc]: Removed

File: src/aq/BackgroundCheck.h [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+#include "oops/util/parameters/RequiredParameter.h"
+
+namespace aq {
+  class GeoVals;
+  template <typename DATATYPE> class ObsData;
+  class ObsSpace;
+  class ObsDiagnostics;
+  class ObsVec;
+
+/// Parameters for L95 BackgroundCheck
```
[the remark this answers is not in the document]
Comment (emanueleemili) [reply src/aq/BackgroundCheck.h]: Done

File: src/aq/aq_constants_mod.F90 [hunk 6]
```diff
@@ -9,6 +9,7 @@ module aq_constants_mod
    public :: oops_int, oops_long, oops_single, oops_real
    public :: aq_int, aq_long, aq_single, aq_real, aq_double
    public :: pi, deg_to_rad, rad_to_deg, req
+   public :: missing_value
```
[the remark this answers is not in the document]
Comment (emanueleemili) [reply src/aq/aq_constants_mod.F90]: Done

File: src/aq/aq_obsvec_mod.F90 (hunk 1 above)
Comment (emanueleemili) [reply src/aq/aq_obsvec_mod.F90]: Removed

File: test/testinput/3densvar.yaml (hunk 2 above)
Comment (emanueleemili) [reply test/testinput/3densvar.yaml]: New test included

**Review by andreapiacentini — APPROVED** [review]

**Pull request merged**